### PR TITLE
Add format checker

### PR DESCRIPTION
**Pull request by abhaybd**
This adds a CI pipeline to check if newly added code adheres to our style guide.

### CODE REVIEW

File: .github/workflows/format-checker.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  build:
+    runs-on: ubuntu-20.04
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v1
+      with:
+        submodules: true
+    - name: Install dependencies from APT
+      run: |
+        sudo apt install git clang-format
```
Comment (huttongrabiel) [inline .github/workflows/format-checker.yml:15]: Should probably specify -y to auto confirm any prompt from apt.

Comment (abhaybd) [reply .github/workflows/format-checker.yml:15]: Oh good catch I'll add that

**Review by huttongrabiel — APPROVED** [review]

**Comment by huttongrabiel** [comment]
lgtm

**Comment by huttongrabiel** [comment]
Non-important to merging this, code looks fine, but you might consider squashing the three commits into one. A lot of the first commit is changed in the second, and the third commit is a two character fix to the second. Not sure how strict you want commit history here, but it could be nice. However, not at all necessary. :+1:

**Comment by abhaybd** [comment]
> Non-important to merging this, code looks good, but you might consider squashing the three commits into one. A lot of the first commit is changed in the second, and the third commit is a two character fix to the second. Not sure how strict you want commit history here, but it could be nice. However, not at all necessary. :+1: 

All PRs are automatically squashed while merging to master, so that will be addressed. It's good you're keeping that in mind though!

**Comment by huttongrabiel** [comment]
> > Non-important to merging this, code looks good, but you might consider squashing the three commits into one. A lot of the first commit is changed in the second, and the third commit is a two character fix to the second. Not sure how strict you want commit history here, but it could be nice. However, not at all necessary. +1
> 
> All PRs are automatically squashed while merging to master, so that will be addressed. 

Good to know!

**Pull request merged**